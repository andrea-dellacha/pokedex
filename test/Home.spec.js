import { shallowMount } from '@vue/test-utils'
import HomePage from '@/pages/index.vue'

describe('HomePage', () => {
  const dispatch = jest.fn((n) => {})

  const wrapper = shallowMount(HomePage, {
    mocks: {
      $store: {
        dispatch,
        state: {
          pokemon: {
            pokemons: [],
            pokemonsQuantity: 0
          }
        }
      }
    },
    computed: {
      pokemons () {
        return this.$store.state.pokemon.pokemons
      },
      pokemonsQuantity () {
        return this.$store.state.pokemon.pokemonsQuantity
      }
    }
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })
})
