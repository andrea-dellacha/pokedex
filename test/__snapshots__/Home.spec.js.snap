// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div
  class="homePage"
>
  <input
    placeholder="Search by pokemon name (only loaded pokemons)"
    type="search"
  />
   
  <div
    class="pokedex"
  />
   
  <div
    class="loader"
  >
    <div
      class="loader__pokeball -duck1"
    />
     
    <div
      class="loader__pokeball -duck2"
    />
     
    <div
      class="loader__pokeball"
    />
     
    <div
      class="loader__pikachu"
    />
     
    <div
      class="loader__ash"
    />
     
    <div
      class="loader__chen"
      style="display: none;"
    />
  </div>
   
  <div
    class="pagination__msg"
  >
    <div
      class="loadMore"
    >
      <loadmoreicon-stub />
    </div>
  </div>
</div>
`;
